framework/src/service/ServiceListeners.cpp

```diff
@@ -446,7 +446,18 @@ void ServiceListeners::GetMatchingServiceListeners(const ServiceEvent& evt,
       const LDAPExpr& ldapExpr = sse.GetLDAPExpr();
       if (ldapExpr.IsNull() || ldapExpr.Evaluate(props, false)) {
         set.insert(sse);
-      }
+      } else if (evt.GetType() == ServiceEvent::SERVICE_REGISTERED) {
+        const std::string msg =
+          "References LDAP filter value in key 'target' of " +
+          sse.GetBundleContext().GetBundle().GetSymbolicName() +
+          " manifest doesn't match service reference properties'." +
+          "\nPlease verify ldap filter value '" + sse.GetLDAPExpr().ToString() +
+          "' in " + ref.GetBundle().GetSymbolicName() +
+          " manifest components 'properties' map.";
```

Comment:
the use of the word "_target_" is specific to declarative services and this log message could be logged for services that are not using declarative services. It would make more sense to the user if the debug message was more generic in terms of what it's trying to tell the user. e.g. describe instead that a service listener matching the LDAP filter could not be found using the LDAP filter _X_.

I'm also not sure the bundle symbolic name is relevant for this debug message. I would suggest leaving it out of the debug message.